src/NServiceBus.Core/Diagnostics/TracingExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace NServiceBus;
+
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Pipeline;
+
+static class TracingExtensions
+{
+    public static Task Invoke<TContext>(this IPipeline<TContext> pipeline, TContext context, Activity activity) where TContext : IBehaviorContext
+    {
+        return activity == null ? pipeline.Invoke(context) : TracePipelineStatus();
+
+        async Task TracePipelineStatus()
+        {
+#pragma warning disable PS0019 // When catching System.Exception, cancellation needs to be properly accounted for
```

Comment:
what would be "properly accounted for" mean in this context?

Reply:
The question is if we want to report `TaskCancelledException` as an error? I think it may make sense as the subsequent calls may not happen and the error status may indicate why.